File: backend/app/schemas/alerts.py
"""
alerts.py
Pydantic schemas for in-app alert notifications.
"""

from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel


class AlertOut(BaseModel):
    id: UUID
    recipient_role: str
    mine_site_id: Optional[UUID] = None
    observation_id: Optional[UUID] = None
    message: str
    is_read: bool
    created_at: datetime

    class Config:
        from_attributes = True

File: dashboard/src/api/alerts.ts
/**
 * Alerts API — wraps GET /alerts and PATCH /alerts/{id}/read
 */
import apiClient from './client';

export interface Alert {
  id: string;
  recipient_role: string;
  mine_site_id: string | null;
  observation_id: string | null;
  message: string;
  is_read: boolean;
  created_at: string;
}

export async function fetchAlerts(unreadOnly = false): Promise<Alert[]> {
  const res = await apiClient.get<Alert[]>('/alerts', {
    params: { unread_only: unreadOnly },
  });
  return res.data;
}

export async function markAlertRead(id: string): Promise<Alert> {
  const res = await apiClient.patch<Alert>(`/alerts/${id}/read`);
  return res.data;
}
